### ISSUE TITLE
some question about the KL loss.

### ISSUE THREAD

**Issue by YanDingXin** [MichiganCOG/TASED-Net#5]
Hello, I am very interested in your work, I have a question to ask.Could you please tell me how your KL-loss is realized?

**Comment by kylemin** [comment]
Hi,

Thank you for your interest.
If you use PyTorch, you can use `torch.nn.KLDivLoss` module.
If not, you can implement your own code calculating KL divergence like below:

```
import sys
import numpy as np
def KLDiv(x, y):
    # x, y are gray-scale images
    x = x/np.sum(x)
    y = y/np.sum(y)
    # now, they represent probability distribution
    eps = sys.float_info.epsilon
    return np.sum(y*np.log(eps+y/(x+eps))))
```

Then, you can get KL-loss by taking the average over the batch.
I hope it helps.
Thank you.

**Comment by YanDingXin** [comment]
Ok, I see. Thank you very much for your reply.
